src/t8_schemes/t8_default/t8_default_pyramid/t8_dpyramid_bits.c

```diff
@@ -1561,25 +1561,32 @@ t8_dpyramid_compute_reference_coords (const t8_dpyramid_t *elem, const double *r
   T8_ASSERT (t8_dpyramid_is_valid (elem));
   if (t8_dpyramid_shape (elem) == T8_ECLASS_PYRAMID) {
     const t8_dpyramid_coord_t length = T8_DPYRAMID_LEN (elem->pyramid.level);
-
-    for (size_t coord = 0; coord < num_coords; ++coord) {
+    size_t coord;
+    for (coord = 0; coord < num_coords; ++coord) {
       const size_t offset = coord * 3;
       out_coords[offset + 0] = elem->pyramid.x;
       out_coords[offset + 1] = elem->pyramid.y;
       out_coords[offset + 2] = elem->pyramid.z;
-
-      if (elem->pyramid.type == T8_DPYRAMID_FIRST_TYPE) {
+    }
+    if (elem->pyramid.type == T8_DPYRAMID_FIRST_TYPE) {
```

Comment:
You could also use `T8_DPYRAMID_FIRST_TYPE` as a default type and only use the else part as a correction term for `T8_DPYRAMID_SECOND_TYPE`. That way we avoid spaghetti-code.

Reply:
done